lib/logger/lib/logger.ex

```diff
@@ -861,7 +905,8 @@ defmodule Logger do
           Logger.Config.compare_levels(level, min_level) == :lt
 
         {k, v} when is_atom(k) ->
-          Keyword.fetch(compile_metadata, k) == {:ok, v}
+          # TODO: Warn on matching on `:module` and `:function` fields
```

Comment:
My suggestion: keep module and function working but as a "metafilter" that looks inside the MFA. We could do the same in the formatter. So we only set mfa as metadata but our APIs know how to look inside the MFA to extract it out. WDYT?

Reply:
Can be done. I just reintroduced the values just for this as it was simpler, but I can define it as a "metafilter" instead. It should probably be even better as it will not compute legacy fields if not needed.

Reply:
@hauleth exactly. I prefer the metafilter so we can start slowly unify on Erlang metadata.